src/GSoft.Extensions.MediatR.Analyzers/Internals/RoslynExtensions.cs

```diff
@@ -0,0 +1,62 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Operations;
+
+namespace GSoft.Extensions.MediatR.Analyzers.Internals;
+
+internal static class RoslynExtensions
+{
+    public static INamedTypeSymbol? GetBestTypeByMetadataName(this Compilation compilation, string typeMetadataName, string? assemblyName = null)
+    {
+        var assemblies = GetAllAssemblies(compilation);
+
+        if (assemblyName != null)
+        {
+            assemblies = assemblies.Where(x => x.Name == assemblyName);
+        }
+
+        foreach (var assembly in assemblies)
+        {
+            if (assembly.GetTypeByMetadataName(typeMetadataName) is { } symbol)
+            {
+                return symbol;
+            }
+        }
+
+        return null;
+    }
+
+    private static IEnumerable<IAssemblySymbol> GetAllAssemblies(Compilation compilation)
```

Comment:
[Nit] Should `public` and `private` methods be grouped together? 

Out of my head, it seems like this would be configured in `.editorconfig` files.

Reply:
I'm not strongly against this, but I really prefer when methods order follow the way they are consumed and read. It's like a book and I don't need an IDE to jump to an implementation.

Especially when I read code on my phone.